### PR TITLE
[ci skip] document compiler/ic/rodfiles.nim

### PR DESCRIPTION
**Pull request by saem**
Why?
* understand how rodfile module works and a bit of the format
* leave notes behind for others
* rather than Araq guess what others need, he can fix what other glean
* possible model for making the compiler more aproachable

Bonus:
* might have found a minor bug in `loadSection`

### CODE REVIEW

**Comment by saem** [comment]
@Araq I was poking around and the navigator bits and started reading `rodfiles.nim`, decided to pretend like I knew the intention and documented it as if answering my questions. I'm hoping it's an easy review where you can check my understanding and I can update appropriately. 🤞🏽 Thanks.

PS. I might have found a minor bug, see the `XXX` comment in `loadSection`.

File: compiler/ic/rodfiles.nim [hunk 1]
```diff
@@ -7,8 +7,65 @@
...
+## * sequences AKA `seq[T]`
+##
+## Note on error handling style
+## ----------------------------
+## A flag based approach is used where operations no-op in case of a
+## preexisting error and set the flag if they encounter one.
+## 
+## Misc
+## ----
+## * 'Prim' is short for 'primitivate', as in a non-sequence type
```
Comment (Araq) [inline compiler/ic/rodfiles.nim]:
```suggestion
## * 'Prim' is short for 'primitive', as in a non-sequence type
```

File: compiler/ic/rodfiles.nim [hunk 2]
```diff
@@ -7,8 +7,65 @@
...
+## sets the `RodFile.err` field. A user of this API needs to handle these
+## appropriately.
+## 
+## API Notes
+## =========
+## 
+## Valid inputs for Rod files
+## --------------------------
+## ASTs, hopes, dreams, and anything as long as it and any children it may have
+## support `copyMem`. At a glance these are:
```
Comment (Araq) [inline compiler/ic/rodfiles.nim]:
```suggestion
## support `copyMem`. This means anything that is not a pointer and that does not contain a pointer. At a glance these are:
```

File: compiler/ic/rodfiles.nim [hunk 3]
```diff
@@ -153,24 +221,29 @@ proc loadHeader*(f: var RodFile) =
...
 proc storeSection*(f: var RodFile; s: RodSection) =
+  ## update `currentSection` and writes the bytes value of s.
   if f.err != ok: return
   assert f.currentSection < s
   f.currentSection = s
   storePrim(f, s)
 
 proc loadSection*(f: var RodFile; expected: RodSection) =
+  ## read the bytes value of s, sets and error if the section is incorrect.
+  ## XXX: should this not update `f.currentSection`?
```
Comment (Araq) [inline compiler/ic/rodfiles.nim]: No, it shouldn't. The caller always knows the current section anyway.

Comment (Araq) [inline compiler/ic/rodfiles.nim]:
```suggestion
```

**Pull request merged**